payments-core/src/main/java/com/stripe/android/payments/core/authentication/PaymentAuthenticator.kt

```diff
@@ -26,5 +30,19 @@ interface PaymentAuthenticator<Authenticatable> : ActivityResultLauncherHost {
         host: AuthActivityStarterHost,
         authenticatable: Authenticatable,
         requestOptions: ApiRequest.Options
+    ) {
+        val lifecycleOwner = host.lifecycleOwner
+        lifecycleOwner.lifecycleScope.launch {
+            lifecycleOwner.whenResumed {
+                performAuthentication(host, authenticatable, requestOptions)
+            }
+        }
```

Comment:
Using this in the public `authenticate()` method and calling a protected method that subclasses can override. Seemed better to me than making every single authenticator implement the lifecycle observation. Thoughts?